Source/Core/CesiumTerrainProvider.js

```diff
@@ -512,10 +554,34 @@ define([
                 encodedNormalBuffer = new Uint8Array(buffer, pos, vertexCount * 2);
             } else if (extensionId === QuantizedMeshExtensionIds.WATER_MASK && provider._requestWaterMask) {
                 waterMaskBuffer = new Uint8Array(buffer, pos, extensionLength);
+            } else if (extensionId === QuantizedMeshExtensionIds.BVH && provider._requestBvh) {
+                var extensionPos = pos;
+                 // Align to 4 bytes.
+                if (extensionPos % 4 !== 0) {
+                    extensionPos += (4 - (extensionPos % 4));
+                }
+
+                var numberOfHeights = view.getUint32(extensionPos, true);
+                extensionPos += Uint32Array.BYTES_PER_ELEMENT;
+                var bvh = new Float32Array(buffer, extensionPos, numberOfHeights);
+                extensionPos += Float32Array.BYTES_PER_ELEMENT * numberOfHeights;
+
+                var numberofIncludedLevels = findNumberOfLevels(numberOfHeights/2);
```

Comment:
The iterative algorithm in `findNumberOfLevels` is pretty expensive to do on every tile, and I don't think it's necessary. For the error checking part, just calculate the number of tiles that should be in the BVH at this level, and see if the actual number of tiles is the same. For `recurseHeights`, you just need to adjust your ranges as you recurse. For the loaded tile, the BVH range is 0 through length-1. Then the number of BVH tiles in each child is n = (length - 1) / 4, the southwest child is 1 through n, southeast child is n + 1 through 2n, northwest is 2n + 1 through 3n, and northeast is 3n + 1 through 4n.